File: LeetCode Questions + Tutorials/EngineeringDigest/BitManipulation.java
public class BitManipulation {
    public static void main(String[] args) {
        System.out.println(decimalToBinary(0));
        System.out.println(binaryToDecimal("101"));

        System.out.println(5 & 6);
        System.out.println(5 | 6);
        System.out.println(5 ^ 6);
        System.out.println(5 << 1);
        System.out.println(5 >> 1);

        System.out.println((34456 & 1) == 1);
    }

    private static String decimalToBinary(int n) {
        String res = "";
        if (n == 0) {
            return "0";
        }
        while (n > 0) {
            int rem = n % 2;
            n = n / 2;
            res += rem;
        }
        return new StringBuilder(res).reverse().toString();
    }

    private static int binaryToDecimal(String bNum) {
        int length = bNum.length();
        int m = 1;
        int res = 0;
        for (int i = length - 1; i >= 0; i--) {
            char lastChar = bNum.charAt(i);
            Integer lastBit = Integer.valueOf(String.valueOf(lastChar));
            res = res + m * lastBit;
            m = m * 2;
        }
        return res;
    }
}

File: LeetCode Questions + Tutorials/EngineeringDigest/BitManipulation.ts
function decimalTOBinary(num: number): string {
    let res: string = "";
    while (num > 0) {
        let rem: number = num % 2;
        res = rem + res;
        num = Math.floor(num / 2);
    }
    return res || "0";
}

function binaryToDecimal(num: string): number {
    let length = num.length;
    let m: number = 1;
    let res: number = 0;
    for (let i = length - 1; i >= 0; i--) {
        let lastChar: string = num.charAt(i);
        let lastBit: number = parseInt(lastChar)
        res = res + m * lastBit;
        m = m * 2;
    }
    return res;
}


console.log(`Binary value of 2: ${decimalTOBinary(2)}`);
console.log(`Decimal value of 101: ${binaryToDecimal("101")}`);

console.log(`5 & 6 = ${5 & 6}`);
console.log(`5 | 6 = ${5 | 6}`);
console.log(`5 ^ 6 = ${5 ^ 6}`);
console.log(`5 << 1 = ${5 << 1}`);
console.log(`5 >> 1 = ${5 >> 1}`);

export { };